Infineon pse84 TF-M support

sreeramIfx opened the pull request:
List of patches already merged into the upstream to enable PSE84 TF-M support:

https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/44462
https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/44178
https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/44379
https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/45386
https://review.trustedfirmware.org/c/TF-M/trusted-firmware-m/+/44519

Anton-TF committed:
[zep fromtree dirty] Build: Add noreturn attribute to specific functions

Marked the following functions with the __NO_RETURN attribute:
 - tfm_hal_system_reset()
 - tfm_hal_system_halt()
 - tfm_core_panic()

This change:
 - reduces compiler warnings
 - enables better optimizations
 - minimizes false positives in static analysis

Signed-off-by: Anton Komlev <anton.komlev@arm.com>
Change-Id: I50a2bfa10cd73d736dd8b13cd9afb37adc4cf03b
(cherry picked from commit 3d698f6cc90e155fe9a28312913cdad3fa147e03)
Signed-off-by: Sreeram Tatapudi <sreeram.praveen@infineon.com>

UEWBot committed:
[zep fromtree] Build: Add support for optional platform post_config.cmake

This is included after all other configuration files have been included.

Change-Id: Ie7f4dfb5238e0ec034a0a3e9b5c82571e6956494
Signed-off-by: Chris Brand <chris.brand@cypress.com>
(cherry picked from commit 776a2072b081178499f5bb656f6b6535dd79d5b6)

sd75t3p committed:
[zep fromtree] Build: Support overriding secure test partition stack size

Check for an existing value for SECURE_TEST_PARTITION_STACK_SIZE
before setting it.

Change-Id: I899c8694e7c47c392f2a0e200e0ce52edbdc5a03
Signed-off-by: Roman Mazurak <roman.mazurak@infineon.com>
Signed-off-by: Chris Brand <chris.brand@cypress.com>
(cherry picked from commit 57bf34a1b12b5eb53807fb71b5be74efe0695da3)

kharkeev committed:
[zep fromtree] Build: Fix PLATFORM_DEFAULT_CRYPTO_KEYS cmake error

In cmake/install.cmake, the install to (and hence creation of)
${INTERFACE_INC_DIR}/crypto_keys is conditioned on
if(PLATFORM_DEFAULT_CRYPTO_KEYS), so the use of that directory in
target_include_directories(tfm_api_ns) also needs the same condition.

This fixes the following error:
  Imported target "psa_crypto_config" includes non-existent path
    ".../api_ns/interface/include/crypto_keys"
  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
  * The path was deleted, renamed, or moved to another location.
  * An install or uninstall procedure did not complete successfully.
  * The installation package was faulty and references files it does not
  provide.

Co-Authored-by: Burian Bohdan <bohdan.burian@infineon.com>
Signed-off-by: Chris Brand <chris.brand@cypress.com>
Change-Id: I2d348e7e327f52664e50a6186b87cb6074d82a73
(cherry picked from commit 593344d096f12a7bbf6b24034bb9f4bf53bf9e59)

UEWBot committed:
[zep fromtree] Platform: Add support for Infineon PSE84 EPC2

Initial support for Infineon's PSE84 EPC2 platform.

Change-Id: Ice1e979ec51dc252de3229afb515e79248b326c7
Signed-off-by: Chris Brand <chris.brand@cypress.com>
Co-Authored-by: Bohdan Hunko <Bohdan.Hunko@infineon.com>
Co-Authored-by: Bohdan Burian <Bohdan.Burian@infineon.com>
Co-Authored-by: Khyzhniak Oleksii <Oleksii.Khyzhniak-EE@infineon.com>
Co-Authored-by: Artur Kremenchuk <Artur.Kremenchuk-EE@infineon.com>
Co-Authored-by: Kostiantyn Tkachov <kostiantyn.tkachov@infineon.com>
Co-Authored-by: Kozemchuk Ivan <ivan.kozemchuk@infineon.com>
Co-Authored-by: Kytsun Hennadiy <hennadiy.kytsun@infineon.com>
Co-Authored-by: Peschanskyi Danyil <danyil.peschanskyi@infineon.com>
Co-Authored-by: Roman Mazurak <roman.mazurak@infineon.com>
Co-Authored-by: Smutniak Oleh <oleh.smutniak@infineon.com>
(cherry picked from commit 78cfcd54a1eee947a34c094ddd46f2b68bd55684)

hunkob committed:
[zep fromtree] SPM: Introduce API to ensure shared metadata section is protected correctly

This api can be used in scenarios where partition metadata section
needs to be RW accessible by SPM only, while ensuring that partitions
have read-only access to it at all other times.

Change-Id: I649506acd548209e49a841aad1b85ffa0d45b0ea
Signed-off-by: Bohdan Hunko <Bohdan.Hunko@infineon.com>
Signed-off-by: Chris Brand <chris.brand@cypress.com>
(cherry picked from commit 282c282c7823e5537ea613e1d5fb4e900b0e92d0)

sreeramIfx commented:
@tomi-font @valeriosetti This is my first PR in the TF-M fork. I see that most commits have a [zep fromtree/fromlist/noup]. Please let me know the convention for commit messages and I will update the commits accordingly

tomi-font commented:
There is a link to the relevant documentation in the README of this repo: https://github.com/zephyrproject-rtos/trusted-firmware-m (just scroll down)
Do you have any suggestion on how to make it more visible?

sreeramIfx commented:
> There is a link to the relevant documentation in the README of this repo: https://github.com/zephyrproject-rtos/trusted-firmware-m (just scroll down) Do you have any suggestion on how to make it more visible?

Thank you for pointing to the documentation, I have updated the commit messages. I was searching the Discord channel for information on this. Adding this to the pinned messages in the #tf-m channel might be helpful.

tomi-font reviewed (changes requested):
You are missing the `cherry picked from commit` line in the commit messages. You need to cherry-pick the commits with `-x`.

sreeramIfx commented:
> You are missing the `cherry picked from commit` line in the commit messages. You need to cherry-pick the commits with `-x`.

I have cherry-picked the commits again with `-x`. Please review

tomi-font commented:
You'll need a Zephyr PR to pull theses changes in and test them in CI.

hunkob committed:
[zep fromtree] IFX: Added missed files & minor updates

Updated IFX platform to reflect latest changes in metadata handling.
Also added "default" design files.

Change-Id: I724000ffca64a7638f0cf7701f435a1af072ac15
Signed-off-by: Bohdan Hunko <Bohdan.Hunko@infineon.com>
Signed-off-by: Kytsun Hennadiy <Hennadiy.Kytsun@infineon.com>
(cherry picked from commit 67ff7d4b7a33de69045b3f4979a369799a6f747f)

tomi-font commented on secure_fw/spm/include/utilities.h:
```diff
@@ -9,12 +9,18 @@
+/*
+ * Accessor function to get the value of the spm_boundary
+ */
+uintptr_t get_spm_boundary(void);
```
Remove this function prototype from the conflict instead of adding it?

sreeramIfx replied:
removed this function

tomi-font commented on secure_fw/spm/include/utilities.h:
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2018-2022, Arm Limited. All rights reserved.
+ * SPDX-FileCopyrightText: Copyright The TrustedFirmware-M Contributors
```
This line change can be removed, doesn't seem to be part of the original commit.

sreeramIfx replied:
reverted the changes to copyright

tomi-font reviewed (commented):
Looks mostly good. Some remarks on the dirty cherry-pick.
